Decred Wallet/Features/Navigation Menu/TabNavigationMenu.swift

```diff
@@ -0,0 +1,126 @@
+            itemView.clipsToBounds = true
+            let itemWidth = self.frame.width / CGFloat(items.count)
+            let leadingAnchor = itemWidth * CGFloat(i)
+            NSLayoutConstraint.activate([
+                itemView.heightAnchor.constraint(equalTo: self.heightAnchor),
+                itemView.leadingAnchor.constraint(equalTo: self.leadingAnchor, constant: leadingAnchor),
+                itemView.topAnchor.constraint(equalTo: self.topAnchor),
+            ])
+        }
+        self.setNeedsLayout()
+        self.layoutIfNeeded()
+        self.activateTab(viewId: self.activeItem)
+    }
+    
+    // Create a custom nav menu item
+    func createTabItem(item: MenuItem) -> UIView {
+        let tabBarItem = UIView(frame: CGRect.zero)
+        
+        let itemTitleLabel = UILabel(frame: CGRect.zero)
+        let itemIconView = UIImageView(frame: CGRect.zero)
+        
+        itemTitleLabel.font = UIFont(name: "Source Sans Pro", size: 13)
+        itemTitleLabel.text = item.displayTitle
+        itemTitleLabel.textColor = UIColor.appColors.darkGray
+        itemTitleLabel.textAlignment = .center
+        itemTitleLabel.translatesAutoresizingMaskIntoConstraints = false
+        itemTitleLabel.clipsToBounds = true
+        
+        itemIconView.image = item.icon!.withRenderingMode(.automatic)
+        itemIconView.translatesAutoresizingMaskIntoConstraints = false
+        itemIconView.clipsToBounds = true
+        
+        tabBarItem.layer.backgroundColor = UIColor.white.cgColor
+        tabBarItem.addSubview(itemIconView)
+        tabBarItem.addSubview(itemTitleLabel)
+        tabBarItem.translatesAutoresizingMaskIntoConstraints = false
+        tabBarItem.clipsToBounds = true
+        
+        let constraints = [
+            itemIconView.heightAnchor.constraint(equalToConstant: 25), // Fixed height for our tab item
+            itemIconView.widthAnchor.constraint(equalToConstant: 25), // Fixed width for our tab item
+            itemIconView.centerXAnchor.constraint(equalTo: tabBarItem.centerXAnchor),
+            itemIconView.topAnchor.constraint(equalTo: tabBarItem.topAnchor, constant: 8), // Position menu item icon 8pts from the top of it's parent view
+            itemIconView.leadingAnchor.constraint(equalTo: tabBarItem.leadingAnchor, constant: 35),
+            itemTitleLabel.heightAnchor.constraint(equalToConstant: 13), // Fixed height for title label
+            itemTitleLabel.widthAnchor.constraint(equalTo: tabBarItem.widthAnchor), // Position label full width across tab bar item
+            itemTitleLabel.topAnchor.constraint(equalTo: itemIconView.bottomAnchor, constant: 4), // Position title label 4pts below item icon
+        ]
+        NSLayoutConstraint.activate(constraints)
+        tabBarItem.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.handleTap))) // Each item should be able to trigger and action on tap
```

Comment:
Add an empty space after setting the constraints.
This line is too long, move the comment to its own line.

Reply:
resolved in 72e13de